passes/src/main/scala/at/forsyte/apalache/tla/passes/imp/SanyParserPassImpl.scala

```diff
@@ -89,25 +95,29 @@ class SanyParserPassImpl @Inject() (
     }
   }
 
+  override def dependencies = Set()
+
+  override def transformations = Set()
+
   override def execute(module: TlaModule): PassResult = {
-    import SourceOption._
-    val src = options.input.source
-    for {
-      rootModule <- src.format match {
-        case Format.Itf  => throw new SanyImporterException("Parsing the ITF format is not supported")
-        case Format.Json => loadFromJsonSource(src)
-        case Format.Tla =>
-          src match {
-            case StringSource(content, aux, _) => loadFromTlaString(content, aux)
-            case FileSource(file, _)           => loadFromTlaFile(file)
-          }
-      }
-      sortedModule <- sortDeclarations(rootModule)
-      _ <- saveLoadedModule(sortedModule)
-    } yield sortedModule
+    try {
+      parseSource(options.input.source)
+    } catch {
+      case err: SanyException =>
+        val msg = err.getMessage()
+        logger.error(s"Parsing error: ${msg}")
+        passFailure(List(msg), ExitCodes.ERROR)
```

Comment:
It think this obsoletes one path in some of the exception adapters, e.g.

https://github.com/informalsystems/apalache/blob/7477b1be3dae41b187298427d2559da3d4bf1f0d/tla-io/src/main/scala/at/forsyte/apalache/tla/imp/ParserExceptionAdapter.scala#L18-L19